app/frontend/courses/review/components/CoursesReview__Editor.res

```diff
@@ -399,14 +395,21 @@ let closeOverlay = (state, courseId, filter) => {
     : RescriptReactRouter.push(path)
 }
 
-let reviewNextButton = (nextSubmission, send, courseId, filter, submissionId, className) => {
+let reviewNextButton = (nextSubmission, filter, nextSubmissionId, className) => {
   ReactUtils.nullIf(
     <button
-      onClick={_ => getNextSubmission(send, courseId, filter, submissionId)}
+      onClick={_ =>
+        switch nextSubmissionId {
+        | Some(nextSubmissionId) =>
+          RescriptReactRouter.push(
+            "/submissions/" ++ nextSubmissionId ++ "/review?" ++ Filter.toQueryString(filter),
+          )
```

Comment:
When `nextSubmissionId` is `None` you will have a button that doesn't do anything; Issues like this arise as you are trying to store the loading status and value separately;